src/components/cylc/graph/Graph.vue

```diff
@@ -0,0 +1,631 @@
+     * @param {cytoscape} instance - the cytoscape instance
+     */
+    setupNavigator (instance) {
+      instance.navigator({
+        container: this.$refs['cytoscape-navigator-overlay'],
+        viewLiveFramerate: 0, // set false to update graph pan only on drag end set 0 to do it instantly set a number (frames per second) to update not more than N times per second
+        thumbnailEventFramerate: 30, // max thumbnail's updates per second triggered by graph updates
+        thumbnailLiveFramerate: false, // max thumbnail's updates per second. Set false to disable
+        dblClickDelay: 200, // milliseconds
+        removeCustomContainer: true, // destroy the container specified by user on plugin destroy
+        rerenderDelay: 100 // ms to throttle rerender updates to the panzoom for performance
+      })
+    },
+
+    /**
+     * Sets up interactivity on the graph node. Defining what happens when the user clicks on nodes, edges, etc.
+     * @param {cytoscape} instance - the cytoscape instance
+     */
+    setupInteractivity (instance) {
+      instance.on('tap', 'node', (event) => {
+        const node = event.target
+        const ref = node.popperRef()
+        this.tippy = new Tippy(ref, {
+          content: () => {
+            const content = document.createElement('div')
+            content.id = 'tooltip'
+            content.classList.add('node-tooltip')
+            let runPercent = 0
+            if (has(node.data, 'runpercent')) {
+              runPercent = node.data('runpercent')
+            }
+            const parent = node.data('parent')
+            const state = node.data('state')
+            const taskSquare = '<div class="box-task-graph ' + state + '-graph"></div>'
+            const jobs = node.data('jobs')
+
+            let jobInfoBlock = '<div>'
+            for (const job of jobs) {
+              const jobId = `<span class="jobid">${job.batchSysJobId}</span>`
+              const jobSquare = `<div class="box-job-graph ${job.state}-graph"></div>`
+              const details = `<details><summary>${jobId}${jobSquare}</summary>`
+              const submitNum = `<br><strong>submit number:  <span style="color: #555;">${job.submitNum}</span></strong>`
+              const batchSysName = `<br><strong>batch sytem name: <span style="color: #555;">${job.batchSysName}</span></strong>`
+              const batchSysJobId = `<br><strong>batch sytem job id: <span style="color: #555;">${job.batchSysJobId}</span></strong>`
+              const host = `<br><strong>host:  <span style="color: #555;">${job.host}</span></strong>`
+              const startedTime = `<br><strong>started time:  <span style="color: #555;">${job.startedTime}</span></strong>`
+              const submittedTime = `<br><strong>submitted time:  <span style="color: #555;">${job.submittedTime}</span></strong>`
+              const finishedTime = `<br><strong>finished time:  <span style="color: #555;">${job.finishedTime}</span></strong><br>`
```

Comment:
Several string concatenation operations replaced by backticks.

However, we still have a lot of `innerHTML` and styles and attributes being changed directly. This is quite bad for performance, possible to cause reflows too.

- https://coderwall.com/p/o9ws2g/why-you-should-always-append-dom-elements-using-documentfragments
- https://www.sitepoint.com/10-ways-minimize-reflows-improve-performance/

Reply:
It's nasty but this code won't make it to 1.0.0 so we can ignore it for now.